docs/configure.md
# 配置文档

## 全部配置

```yml
# serverless.yml

component: nextjs # (必选) 组件名称，在该实例中为nextjs
name: nextjsDemo # 必选) 组件实例名称.
org: orgDemo # (可选) 用于记录组织信息，默认值为您的腾讯云账户 appid，必须为字符串
app: appDemo # (可选) 用于记录组织信息. 默认与name相同，必须为字符串
stage: dev # (可选) 用于区分环境信息，默认值是 dev

inputs:
  region: ap-guangzhou # 云函数所在区域
  functionName: nextjsDemo # 云函数名称
  serviceName: mytest # api网关服务名称
  runtime: Nodejs10.15 # 运行环境
  serviceId: service-np1uloxw # api网关服务ID
  entryFile: sls.js # 自定义 server 的入口文件名，默认为 sls.js，如果不想修改文件名为 sls.js 可以自定义
  src: ./src # 第一种为string时，会打包src对应目录下的代码上传到默认cos上。
  # src:  # 第二种，部署src下的文件代码，并打包成zip上传到bucket上
  #   src: ./src  # 本地需要打包的文件目录
  #   bucket: bucket01 # bucket name，当前会默认在bucket name后增加 appid 后缀, 本例中为 bucket01-appid
  #   exclude:   # 被排除的文件或目录
  #     - .env
  #     - node_modules
  # src: # 第三种，在指定存储桶bucket中已经存在了object代码，直接部署
  #   bucket: bucket01 # bucket name，当前会默认在bucket name后增加 appid 后缀, 本例中为 bucket01-appid
  #   object: cos.zip  # bucket key 指定存储桶内的文件
  layers:
    - name: layerName #  layer名称
      version: 1 #  版本
  functionConf: # 函数配置相关
    timeout: 10 # 超时时间，单位秒
    eip: false # 是否固定出口IP
    memorySize: 128 # 内存大小，单位MB
    environment: #  环境变量
      variables: #  环境变量数组
        TEST: vale
    vpcConfig: # 私有网络配置
      vpcId: '' # 私有网络的Id
      subnetId: '' # 子网ID
  apigatewayConf: #  api网关配置
    isDisabled: false # 是否禁用自动创建 API 网关功能
    isBase64Encoded: false # 是否开启 base64 编码
    enableCORS: true #  允许跨域
    customDomains: # 自定义域名绑定
      - domain: abc.com # 待绑定的自定义的域名
        certificateId: abcdefg # 待绑定自定义域名的证书唯一 ID
        # 如要设置自定义路径映射，请设置为 false
        isDefaultMapping: false
        # 自定义路径映射的路径。使用自定义映射时，可一次仅映射一个 path 到一个环境，也可映射多个 path 到多个环境。并且一旦使用自定义映射，原本的默认映射规则不再生效，只有自定义映射路径生效。
        pathMappingSet:
          - path: /
            environment: release
        protocols: # 绑定自定义域名的协议类型，默认与服务的前端协议一致。
          - http # 支持http协议
          - https # 支持https协议
    protocols:
      - http
      - https
    environment: test
    serviceTimeout: 15
    usagePlan: #  用户使用计划
      usagePlanId: 1111
      usagePlanName: slscmp
      usagePlanDesc: sls create
      maxRequestNum: 1000
    auth: #  密钥
      secretName: secret
      secretIds:
        - xxx
  staticConf:
    cosConf:
      bucket: static-bucket
      sources:
        - src: .next/static
          targetDir: /_next/static
        - src: public
          targetDir: /
    cdnConf:
      area: mainland
      domain: cnode.yuga.chat
      autoRefresh: true
      refreshType: delete
      forceRedirect:
        switch: on
        redirectType: https
        redirectStatusCode: 301
      https:
        http2: on
        certId: 'abc'
```

## 配置描述

主要的参数

| 参数名称                             | 必选 |     默认值      | 描述                                                            |
| ------------------------------------ | :--: | :-------------: | :-------------------------------------------------------------- |
| runtime                              |  否  |  `Nodejs10.15`  | 执行环境, 支持版本参考：[目前维护版本](https://cloud.tencent.com/document/product/583/11060)  |
| region                               |  否  | `ap-guangzhou`  | 项目部署所在区域                                                |
| functionName                         |  否  |                 | 云函数名称                                                      |
| serviceName                          |  否  |                 | API 网关服务名称, 默认创建一个新的服务名称                      |
| serviceId                            |  否  |                 | API 网关服务 ID, 如果存在将使用这个 API 网关服务                |
| entryFile                            |  否  |    `sls.js`     | 自定义 server 的入口文件名                                      |
| src                                  |  否  | `process.cwd()` | 默认为当前目录, 如果是对象, 配置参数参考 [执行目录](#执行目录)  |
| layers                               |  否  |                 | 云函数绑定的 layer, 配置参数参考 [层配置](#层配置)              |
| [functionConf](#函数配置)            |  否  |                 | 函数配置                                                        |
| [apigatewayConf](#API-网关配置)      |  否  |                 | API 网关配置                                                    |
| [cloudDNSConf](#DNS-配置)            |  否  |                 | DNS 配置                                                        |
| [Region special config](#指定区配置) |  否  |                 | 指定区配置                                                      |
| [staticConf](#静态资源-CDN-配置)     |  否  |                 | 静态资源 CDN 配置                                               |

## 执行目录

| 参数名称 | 是否必选 |   类型   | 默认值 | 描述                                                                                                                                                                                 |
| -------- | :------: | :------: | :----: | :----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| src      |    否    |  String  |        | 代码路径。与 object 不能同时存在。                                                                                                                                                   |
| exclude  |    否    | String[] |        | 不包含的文件或路径, 遵守 [glob 语法](https://github.com/isaacs/node-glob)                                                                                                            |
| bucket   |    否    |  String  |        | bucket 名称。如果配置了 src，表示部署 src 的代码并压缩成 zip 后上传到 bucket-appid 对应的存储桶中；如果配置了 object，表示获取 bucket-appid 对应存储桶中 object 对应的代码进行部署。 |
| object   |    否    |  String  |        | 部署的代码在存储桶中的路径。                                                                                                                                                         |

## 层配置

| 参数名称 | 是否必选 |  类型  | 默认值 | 描述     |
| -------- | :------: | :----: | :----: | :------- |
| name     |    否    | String |        | 层名称   |
| version  |    否    | String |        | 层版本号 |

### DNS 配置

参考: https://cloud.tencent.com/document/product/302/8516

| 参数名称   | 是否必选 | 类型     | 默认值 | 描述                                            |
| ---------- | :------: | -------- | :----: | :---------------------------------------------- |
| ttl        |    否    | Number   | `600`  | TTL 值，范围 1 - 604800，不同等级域名最小值不同 |
| recordLine |    否    | String[] |        | 记录的线路名称                                  |

### 指定区配置

| 参数名称                        | 是否必选 | 类型   | 默认值 | 函数         |
| ------------------------------- | :------: | ------ | ------ | ------------ |
| [functionConf](#函数配置)       |    否    | Object |        | 函数配置     |
| [apigatewayConf](#API-网关配置) |    否    | Object |        | API 网关配置 |
| [cloudDNSConf](#DNS-配置)       |    否    | Object |        | DNS 配置     |

### 函数配置

参考: https://cloud.tencent.com/document/product/583/18586

| 参数名称    | 是否必选 |  类型   | 默认值  | 描述                                                                            |
| ----------- | :------: | :-----: | :-----: | :------------------------------------------------------------------------------ |
| timeout     |    否    | Number  |   `3`   | 函数最长执行时间，单位为秒，可选值范围 1-900 秒，默认为 3 秒                    |
| memorySize  |    否    | Number  |  `128`  | 函数运行时内存大小，默认为 128M，可选范围 64、128MB-3072MB，并且以 128MB 为阶梯 |
| environment |    否    | Object  |         | 函数的环境变量, 参考 [环境变量](#环境变量)                                      |
| vpcConfig   |    否    | Object  |         | 函数的 VPC 配置, 参考 [VPC 配置](#VPC-配置)                                     |
| eip         |    否    | Boolean | `false` | 是否固定出口 IP                                                                 |

##### 环境变量

| 参数名称  | 类型 | 描述                                      |
| --------- | ---- | :---------------------------------------- |
| variables |      | 环境变量参数, 包含多对 key-value 的键值对 |

##### VPC 配置

| 参数名称 | 类型   | 描述    |
| -------- | ------ | :------ |
| subnetId | String | 子网 ID |
| vpcId    | String | VPC ID  |

### API 网关配置

| 参数名称        | 是否必选 | 类型     | 默认值     | 描述                                                                               |
| --------------- | :------: | :------- | :--------- | :--------------------------------------------------------------------------------- |
| protocols       |    否    | String[] | `['http']` | 前端请求的类型，如 http，https，http 与 https                                      |
| environment     |    否    | String   | `release`  | 发布环境. 目前支持三种发布环境: test（测试）, prepub（预发布） 与 release（发布）. |
| usagePlan       |    否    |          |            | 使用计划配置, 参考 [使用计划](#使用计划)                                           |
| auth            |    否    |          |            | API 密钥配置, 参考 [API 密钥](#API-密钥配置)                                       |
| customDomain    |    否    | Object[] |            | 自定义 API 域名配置, 参考 [自定义域名](#自定义域名)                                |
| enableCORS      |    否    | Boolean  | `false`    | 开启跨域。默认值为否。                                                             |
| serviceTimeout  |    否    | Number   | `15`       | Api 超时时间，单位: 秒                                                             |
| isDisabled      |    否    | Boolean  | `false`    | 关闭自动创建 API 网关功能。默认值为否，即默认自动创建 API 网关。                   |
| isBase64Encoded |    否    | Boolean  | `false`    | 是否开启 Base64 编码，如果需要文件上传，请配置为 `true`                            |

##### 使用计划

参考: https://cloud.tencent.com/document/product/628/14947

| 参数名称      | 是否必选 | 类型   | 描述                                                    |
| ------------- | :------: | ------ | :------------------------------------------------------ |
| usagePlanId   |    否    | String | 用户自定义使用计划 ID                                   |
| usagePlanName |    否    | String | 用户自定义的使用计划名称                                |
| usagePlanDesc |    否    | String | 用户自定义的使用计划描述                                |
| maxRequestNum |    否    | Number | 请求配额总数，如果为空，将使用-1 作为默认值，表示不开启 |

##### API 密钥配置

参考: https://cloud.tencent.com/document/product/628/14916

| 参数名称   | 类型   | 描述     |
| ---------- | :----- | :------- |
| secretName | String | 密钥名称 |
| secretIds  | String | 密钥 ID  |

##### 自定义域名

Refer to: https://cloud.tencent.com/document/product/628/14906

| 参数名称         | 是否必选 |   类型   | 默认值 | 描述                                                                                                                                                                                 |
| ---------------- | :------: | :------: | :----: | :----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| domain           |    是    |  String  |        | 待绑定的自定义的域名。                                                                                                                                                               |
| certificateId    |    否    |  String  |        | 待绑定自定义域名的证书唯一 ID，如果设置了 type 为 https，则为必选                                                                                                                    |
| isDefaultMapping |    否    |  String  | `true` | 是否使用默认路径映射。为 false 时，表示自定义路径映射，此时 pathMappingSet 必填。                                                                                                    |
| pathMappingSet   |    否    | Object[] |  `[]`  | 自定义路径映射的路径。使用自定义映射时，可一次仅映射一个 path 到一个环境，也可映射多个 path 到多个环境。并且一旦使用自定义映射，原本的默认映射规则不再生效，只有自定义映射路径生效。 |
| protocol         |    否    | String[] |        | 绑定自定义域名的协议类型，默认与服务的前端协议一致。                                                                                                                                 |

- 自定义路径映射

| 参数名称    | 是否必选 | 类型   | Description    |
| ----------- | :------: | :----- | :------------- |
| path        |    是    | String | 自定义映射路径 |
| environment |    是    | String | 自定义映射环境 |

### 静态资源 CDN 配置

| 参数名称 | 是否必选 |  类型  | 默认值 | 描述                  |
| -------- | :------: | :----: | :----: | :-------------------- |
| cosConf  |    是    | Object |        | [COS 配置](#cos-配置) |
| cdnConf  |    否    | Object |        | [CDN 配置](#cdn-配置) |

##### COS 配置

| 参数名称 | 是否必选 |   类型   |                                         默认值                                         | 描述                           |
| -------- | :------: | :------: | :------------------------------------------------------------------------------------: | :----------------------------- |
| bucket   |    是    |  string  |                                                                                        | COS 存储同名称，没有将自动创建 |
| (#acl)   |
| sources  |    否    | Object[] | `[{src: '.next/static', targetDir: '/_next/static'}, {src: 'public', targetDir: '/'}]` | 需要托管到 COS 的静态资源目录  |

##### CDN 配置

area: mainland domain: cnode.yuga.chat autoRefresh: true refreshType: delete
forceRedirect: switch: on redirectType: https redirectStatusCode: 301 https:
http2: on certId: 'eGkM75xv'

| 参数名称      | 是否必选 |  类型   |   默认值   | 描述                                                       |
| ------------- | :------: | :-----: | :--------: | :--------------------------------------------------------- |
| domain        |    是    | string  |            | CDN 域名                                                   |
| area          |    否    | string  | `mainland` | 加速区域，mainland: 大陆，overseas：海外，global：全球加速 |
| autoRefresh   |    否    | boolean |   `true`   | 是否自动刷新 CDN                                           |
| refreshType   |    否    | boolean |  `delete`  | CDN 刷新类型，delete：刷新全部资源，flush：刷新变更资源    |
| forceRedirect |    否    | Object  |            | 访问协议强制跳转配置，参考 [forceRedirect](#forceRedirect) |
| https         |    否    | Object  |            | https 配置，参考 [https](#https)                           |

###### forceRedirect

| 参数名称           | 是否必选 |  类型  | 默认值 | 描述                                                           |
| ------------------ | :------: | :----: | :----: | :------------------------------------------------------------- |
| switch             |    是    | string |  `on`  | 访问强制跳转配置开关, on：开启，off：关闭                      |
| redirectType       |    是    | string | `http` | 访问强制跳转类型，http：强制 http 跳转，https：强制 https 跳转 |
| redirectStatusCode |    是    | number | `301`  | 强制跳转时返回状态码，支持 301、302                            |

###### https

| 参数名称 | 是否必选 |  类型  | 默认值 | 描述                                  |
| -------- | :------: | :----: | :----: | :------------------------------------ |
| certId   |    是    | string |        | 腾讯云托管域名证书 ID                 |
| http2    |    是    | string |        | 是否开启 HTTP2，on： 开启，off： 关闭 |


CHANGELOG.md
## [0.2.1](https://github.com/serverless-components/tencent-nextjs/compare/v0.2.0...v0.2.1) (2021-01-26)


### Bug Fixes

* change cos access by policy ([677c524](https://github.com/serverless-components/tencent-nextjs/commit/677c524868b736ea9bc839e1e235c4d46d58f780))

# [0.2.0](https://github.com/serverless-components/tencent-nextjs/compare/v0.1.10...v0.2.0) (2021-01-26)


### Features

* support apigw base64 encode ([444567e](https://github.com/serverless-components/tencent-nextjs/commit/444567e129da21d3440849e037df49e6925b6f76))

## [0.1.10](https://github.com/serverless-components/tencent-nextjs/compare/v0.1.9...v0.1.10) (2020-12-21)


### Bug Fixes

* update remove flow ([#18](https://github.com/serverless-components/tencent-nextjs/issues/18)) ([a343607](https://github.com/serverless-components/tencent-nextjs/commit/a34360709ee03c35027a349956a7ed59b56464ed))

## [0.1.9](https://github.com/serverless-components/tencent-nextjs/compare/v0.1.8...v0.1.9) (2020-12-15)


### Bug Fixes

* update to deployment to serial flow ([f96518a](https://github.com/serverless-components/tencent-nextjs/commit/f96518a4f2f3254f7b7c2c74e8cb3e383e176595))

## [0.1.8](https://github.com/serverless-components/tencent-nextjs/compare/v0.1.7...v0.1.8) (2020-10-20)


### Bug Fixes

* support replace deployment and cos assets ([85e9b7c](https://github.com/serverless-components/tencent-nextjs/commit/85e9b7c7dea22fb1eb991de4aca5cfcedc6035a7))

## [0.1.7](https://github.com/serverless-components/tencent-nextjs/compare/v0.1.6...v0.1.7) (2020-10-12)


### Bug Fixes

* support all parameters for apigw ([5e22cf3](https://github.com/serverless-components/tencent-nextjs/commit/5e22cf3595c1adfcf84de008d7af94bfc1bc4184))

## [0.1.6](https://github.com/serverless-components/tencent-nextjs/compare/v0.1.5...v0.1.6) (2020-10-10)


### Bug Fixes

* support all configs for scf ([52b8f51](https://github.com/serverless-components/tencent-nextjs/commit/52b8f51b1cebcf8d16a658454256f4f6014f7e0b))

## [0.1.5](https://github.com/serverless-components/tencent-nextjs/compare/v0.1.4...v0.1.5) (2020-09-29)


### Bug Fixes

* support customize sls entry file ([eb5d6b3](https://github.com/serverless-components/tencent-nextjs/commit/eb5d6b3c2406ce7a225296d3e541f1efda029dd5))

## [0.1.4](https://github.com/serverless-components/tencent-nextjs/compare/v0.1.3...v0.1.4) (2020-09-25)


### Bug Fixes

* support koa server ([fd37c09](https://github.com/serverless-components/tencent-nextjs/commit/fd37c09412b5e123840e841289404055814d5bec))

## [0.1.3](https://github.com/serverless-components/tencent-nextjs/compare/v0.1.2...v0.1.3) (2020-09-23)


### Bug Fixes

* update deps ([85ce058](https://github.com/serverless-components/tencent-nextjs/commit/85ce05849638ef0a318915aa39cdbc5a6ae88133))

## [0.1.2](https://github.com/serverless-components/tencent-nextjs/compare/v0.1.1...v0.1.2) (2020-09-03)


### Bug Fixes

* update deps ([b98c0f6](https://github.com/serverless-components/tencent-nextjs/commit/b98c0f653b58fb9a45f1a6baaa5e0a70e47d3b12))

## [0.1.1](https://github.com/serverless-components/tencent-nextjs/compare/v0.1.0...v0.1.1) (2020-09-01)


### Bug Fixes

* update tencent-component-toolkit ([7642596](https://github.com/serverless-components/tencent-nextjs/commit/7642596228f2972cd857096f4d30a31483175175))

# [0.1.0](https://github.com/serverless-components/tencent-nextjs/compare/v0.0.14...v0.1.0) (2020-08-28)


### Bug Fixes

* update deps for uniform error message ([b2a0ea4](https://github.com/serverless-components/tencent-nextjs/commit/b2a0ea423e89a1ec4694bb8fae11b881412b6f36))


### Features

* optimize deploy flow and update deps ([6c34a4d](https://github.com/serverless-components/tencent-nextjs/commit/6c34a4d1d8f74be5f55aa7a1659f5786b366771b))

## [0.0.14](https://github.com/serverless-components/tencent-nextjs/compare/v0.0.13...v0.0.14) (2020-08-26)


### Bug Fixes

* add no report for favicon ([28484c4](https://github.com/serverless-components/tencent-nextjs/commit/28484c48e6011004df81b1177dd19c03f4b077a1))
* apigw custom domain update bug ([9f856b5](https://github.com/serverless-components/tencent-nextjs/commit/9f856b565e67a4656d041e5f0d277f759f3c01ff))
* apigw metics x timestamp ([d68a581](https://github.com/serverless-components/tencent-nextjs/commit/d68a581e02d267871191d16d010c34f1798b0735))
* cache http server ([8e01009](https://github.com/serverless-components/tencent-nextjs/commit/8e01009823fc1682f75f17e7061b9934b03d5f38))
* change staticCdn to staticConf ([e481a94](https://github.com/serverless-components/tencent-nextjs/commit/e481a9492f1c9f7dba0c168fce0be73fa61cd0f4))
* component name ([2f45508](https://github.com/serverless-components/tencent-nextjs/commit/2f455081075379f209600a200b44bd91a50d2658))
* lambdaArn value ([0f63bda](https://github.com/serverless-components/tencent-nextjs/commit/0f63bdae851f6b3ef14c141ee23fd4d2507f54bb))
* read bucket and object from srcOriginal ([c57ae19](https://github.com/serverless-components/tencent-nextjs/commit/c57ae19dd1457d6554fa0b6fc92a6984fc474262))
* static cos has suffix appid ([1bf7400](https://github.com/serverless-components/tencent-nextjs/commit/1bf74008c729ca927e9fb2a201340606f9e8d6e3))
* support apigw serviceTimeout ([3afd507](https://github.com/serverless-components/tencent-nextjs/commit/3afd507a4b72bf5113322ead4d950df6a3a717f2))
* support eip config ([fa0358b](https://github.com/serverless-components/tencent-nextjs/commit/fa0358b85f2ca9af1abe86e1c815c7d9e3ef63ac))
* throw error when no temp secrets ([fe0b6b0](https://github.com/serverless-components/tencent-nextjs/commit/fe0b6b061bec839db8022d3d0f36fee89dd7a146))
* traffic zero display bug ([6e37e82](https://github.com/serverless-components/tencent-nextjs/commit/6e37e824e6e6009dc1fb952ce3e4ed1b38eb8564))
* typo ([2d26ef7](https://github.com/serverless-components/tencent-nextjs/commit/2d26ef732a97f3e56ef9eb96d6b849fea76eaff4))

src/config.js
Object.defineProperty(exports, '__esModule', { value: true })
exports.getConfig = void 0
const fs = require('fs')
const path = require('path')
const YAML = require('js-yaml')
const TEMPLATE_BASE_URL = 'https://serverless-templates-1300862921.cos.ap-beijing.myqcloud.com'
const frameworks = {
  express: {
    injectSlsSdk: true,
    runtime: 'Nodejs10.15',
    defaultEntryFile: 'sls.js',
    defaultStatics: [{ src: 'public', targetDir: '/' }]
  },
  koa: {
    injectSlsSdk: true,
    runtime: 'Nodejs10.15',
    defaultEntryFile: 'sls.js',
    defaultStatics: [{ src: 'public', targetDir: '/' }]
  },
  egg: {
    injectSlsSdk: true,
    runtime: 'Nodejs10.15',
    defaultEntryFile: 'sls.js',
    defaultStatics: [{ src: 'public', targetDir: '/' }],
    defaultEnvs: [
      {
        key: 'SERVERLESS',
        value: '1'
      },
      {
        key: 'EGG_APP_CONFIG',
        value: '{"rundir":"/tmp","logger":{"dir":"/tmp"}}'
      }
    ]
  },
  nestjs: {
    injectSlsSdk: true,
    runtime: 'Nodejs10.15',
    defaultEntryFile: 'sls.js',
    defaultStatics: [{ src: 'public', targetDir: '/' }]
  },
  nextjs: {
    injectSlsSdk: true,
    runtime: 'Nodejs10.15',
    defaultEntryFile: 'sls.js',
    defaultStatics: [
      { src: '.next/static', targetDir: '/_next/static' },
      { src: 'public', targetDir: '/' }
    ]
  },
  nuxtjs: {
    injectSlsSdk: true,
    runtime: 'Nodejs10.15',
    defaultEntryFile: 'sls.js',
    defaultStatics: [
      { src: '.nuxt/dist/client', targetDir: '/' },
      { src: 'static', targetDir: '/' }
    ]
  },
  laravel: {
    injectSlsSdk: false,
    runtime: 'Php7',
    defaultEnvs: [
      {
        key: 'SERVERLESS',
        value: '1'
      },
      {
        key: 'VIEW_COMPILED_PATH',
        value: '/tmp/storage/framework/views'
      },
      {
        key: 'SESSION_DRIVER',
        value: 'array'
      },
      {
        key: 'LOG_CHANNEL',
        value: 'stderr'
      },
      {
        key: 'APP_STORAGE',
        value: '/tmp/storage'
      }
    ]
  },
  thinkphp: {
    injectSlsSdk: false,
    runtime: 'Php7'
  },
  flask: {
    injectSlsSdk: false,
    runtime: 'Python3.6'
  },
  django: {
    injectSlsSdk: false,
    runtime: 'Python3.6'
  }
}
const CONFIGS = {
  // support metrics frameworks
  pythonFrameworks: ['flask', 'django'],
  supportMetrics: ['express', 'next', 'nuxt'],
  region: 'ap-guangzhou',
  description: 'Created by Serverless Component',
  handler: 'sl_handler.handler',
  timeout: 10,
  memorySize: 128,
  namespace: 'default',
  defaultEnvs: [
    {
      key: 'SERVERLESS',
      value: '1'
    }
  ],
  cos: {
    lifecycle: [
      {
        status: 'Enabled',
        id: 'deleteObject',
        expiration: { days: '10' },
        abortIncompleteMultipartUpload: { daysAfterInitiation: '10' }
      }
    ]
  },
  cdn: {
    forceRedirect: {
      switch: 'on',
      redirectType: 'https',
      redirectStatusCode: 301
    },
    https: {
      switch: 'on',
      http2: 'on'
    }
  },
  defaultCdnConfig: {
    forceRedirect: {
      switch: 'on',
      redirectType: 'https',
      redirectStatusCode: 301
    },
    https: {
      switch: 'on',
      http2: 'on'
    }
  },
  acl: {
    permissions: 'public-read',
    grantRead: '',
    grantWrite: '',
    grantFullControl: ''
  },
  getPolicy(region, bucket, appid) {
    return {
      Statement: [
        {
          Principal: { qcs: ['qcs::cam::anyone:anyone'] },
          Effect: 'Allow',
          Action: [
            'name/cos:HeadBucket',
            'name/cos:ListMultipartUploads',
            'name/cos:ListParts',
            'name/cos:GetObject',
            'name/cos:HeadObject',
            'name/cos:OptionsObject'
          ],
          Resource: [`qcs::cos:${region}:uid/${appid}:${bucket}/*`]
        }
      ],
      version: '2.0'
    }
  }
}
const getConfig = () => {
  const { name: framework } = YAML.load(
    // framework.yml 会在组件部署流程中动态生成
    fs.readFileSync(path.join(__dirname, 'framework.yml'), 'utf-8')
  )
  const templateUrl = `${TEMPLATE_BASE_URL}/${framework}-demo.zip`
  const frameworkConfigs = frameworks[framework]
  return Object.assign(Object.assign({ framework, templateUrl }, CONFIGS), frameworkConfigs)
}
exports.getConfig = getConfig
